contents/docs/tutorials/feature-flags.md

```diff
@@ -0,0 +1,206 @@
+
+```js
+if (posthog.isFeatureEnabled('new-beta-feature')) {
+    // do something
+}
+```
+
+This snippet refers to our [JavaScript Integration](/docs/integrations/js-integration), which is also what we will be using for this tutorial. However, you can also use feature flags via our [API](/docs/api/overview) as well as other libraries, such as our [Python Integration](/docs/integrations/python-integration) (we're working to add this functionality to as many of our libraries as possible). 
+
+With that snippet, you can then do whatever you wish inside of it. You might change the CSS of a button, hide an entire section, or move things around. A simple example implementation would be:
+
+```js
+// Header text is blue by default
+// If flag is on, make the text red
+if (posthog.isFeatureEnabled('red-header-text')) {
+    document.getElementById('header-text').style['color'] = 'red'
+}
+```
+
+While this is a trivial example, you can have as much logic as you want being powered by the feature flag. You can also combine flags with `OR` & `AND` operations to create more complex logic.
+
+## Advanced Controls
+
+### Loading Flags
+
+Every time a user loads a page we send a request in the background to an endpoint to get the feature flags that apply to that user. We store those flags as a cookie. 
+
+This means that for most page views the feature flags will be available immediately, *except* for the very first time a user visits your site.
+
+To combat that, there's a callback you can use to wait for the flags to come in:
+
+```js
+posthog.onFeatureFlags(function() {
+    // feature flags are guaranteed to be available at this point
+    if (posthog.isFeatureEnabled('new-beta-feature')) {
+        // do something
+    }
+})
+```
+
+In our JS Integration, the reason we store flags as a cookie is to reduce the load on the server and improve the performance of your app, as it doesn't need to always make an HTTP request, it can simply refer to data stored locally in the browser. 
+
+While this makes your app faster, it means that if your user does something mid-session which causes the flag to turn on for them, this will not be immediately updated. As such, if you expect your app to have scenarios like this _and_ you want flags to update mid-session, you can reload them yourself, by using the `reloadFeatureFlags` function.
+
+```js
+posthog.reloadFeatureFlags()
+```
+
+Calling this function will force PostHog to hit the endpoint for the updated information, and allows you ensure changes are reflected mid-session.
+
+## Feature Flag Uses
+
+Feature flags are a very powerful piece of functionality that can be used in a wide variety of ways. How you use them will depend on your particular painpoints and internal best practices.
+
+For us, here are some suggestions of use cases that could fit nicely with feature flags:
+
+### A/B Testing
+
+![A/B Testing Example](../../images/tutorials/feature-flags/ab-testing.png)
```

Comment:
Would be nicer if this graph was based on an actual PostHog graph (design/style-wise)